sdk/python/kfp/components/component_factory.py

```diff
@@ -266,9 +266,12 @@ def extract_component_interface(
                 output_names.add(output_name)
                 type_var = field_annotations.get(field_name)
                 if type_annotations.is_list_of_artifacts(type_var):
-                    raise ValueError(
-                        f'Cannot use output lists of artifacts in NamedTuple return annotations. Got output list of artifacts annotation for NamedTuple field `{field_name}`.'
-                    )
+                    artifact_cls = type_var.__args__[0]
+                    output_spec = structures.OutputSpec(
+                        type=type_utils.create_bundled_artifact_type(
+                            artifact_cls.schema_title,
+                            artifact_cls.schema_version),
+                        is_artifact_list=True)
```

Comment:
We should support lists of artifacts in `NamedTuple` returns, as users can include a `dsl.Collected` in a `NamedTuple`.